symengine/dense_matrix.cpp

```diff
@@ -1520,6 +1520,57 @@ void inverse_gauss_jordan(const DenseMatrix &A, DenseMatrix &B)
     fraction_free_gauss_jordan_solve(A, e, B);
 }
 
+// ----------------------- Vector-specific Methods --------------------------//
+
+void dot(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C)
+{
+    DenseMatrix tmp1 = DenseMatrix(A.row_, A.col_);
+    DenseMatrix tmp2 = DenseMatrix(B.row_, B.col_);
+
+    for (unsigned i = 0; i < A.row_; i++) {
+        for (unsigned j = 0; j < A.col_; j++) {
+            tmp1.m_[i * A.col_ + j] = A.m_[i * A.col_ + j];
+        }
+    }
+
+    for (unsigned i = 0; i < B.row_; i++) {
+        for (unsigned j = 0; j < B.col_; j++) {
+            tmp2.m_[i * B.col_ + j] = B.m_[i * B.col_ + j];
+        }
+    }
```

Comment:
Can you remove the above unnecessary copying and use them only if they are needed?